adserver/views.py

```diff
@@ -446,6 +451,112 @@ def get_success_url(self):
+        return super().dispatch(request, *args, **kwargs)
+
+    def get_form_kwargs(self):
+        kwargs = super().get_form_kwargs()
+        kwargs["advertiser"] = self.advertiser
+        kwargs["flight"] = self.old_flight
+        return kwargs
+
+    def form_valid(self, form):
+        result = super().form_valid(form)
+        flight = self.object
+
+        # Notify support of the new flight
+        self.send_support_request(
+            flight,
+            {
+                "budget": form.cleaned_data["budget"],
+                "note": form.cleaned_data["note"],
+            },
+        )
+
+        messages.success(
+            self.request,
+            _(
+                "Successfully setup a new (non-live) %(flight)s and notified your account manager"
+            )
+            % {"flight": flight},
+        )
+        return result
+
+    def get_context_data(self, **kwargs):
+        context = super().get_context_data(**kwargs)
+
+        context.update(
+            {
+                "advertiser": self.advertiser,
+                "past_flights": Flight.objects.filter(
+                    campaign__advertiser=self.advertiser
+                ).order_by("-start_date")[:50],
+                "old_flight": self.old_flight,
+                "next": self.request.GET.get("next"),
+            }
+        )
+
+        return context
+
+    def get_success_url(self):
+        return reverse(
+            "flight_detail",
+            kwargs={
+                "advertiser_slug": self.advertiser.slug,
+                "flight_slug": self.object.slug,
+            },
+        )
+
+    def get_old_flight_or_404(self, old_flight_slug):
+        if old_flight_slug:
+            return get_object_or_404(
+                Flight, slug=old_flight_slug, campaign__advertiser=self.advertiser
+            )
+        return None
+
+    def send_support_request(self, new_flight, extras):
+
+        to_email = settings.ADSERVER_SUPPORT_TO_EMAIL
+        if not to_email:
+            site = get_current_site(None)
+            to_email = f"support@{site.domain}"
+            log.warning(
+                "Using the default support email address because ADSERVER_SUPPORT_TO_EMAIL is not configured"
+            )
+
+        context = {
+            "site_domain": generate_absolute_url(""),
+            "user": self.request.user,
+            "advertiser": self.advertiser,
+            "flight": new_flight,
+            "old_flight": self.old_flight,
+            "extras": extras,
+        }
+
+        mail.send_mail(
+            _("New Flight Request - %(advertiser)s")
+            % {"advertiser": self.advertiser.name},
+            message="See HTML message",
+            from_email=settings.DEFAULT_FROM_EMAIL,
+            recipient_list=[to_email],
+            html_message=render_to_string(
+                "adserver/email/flight-request.html", context
+            ),
+        )
```

Comment:
Do we actually want to send email here, or is this creating an object in Front dynamically? I'm fine with either approach for now, but not depending on email here could be a nice benefit eventually, similar to the inbound forms.

Reply:
I'm.... not sure. This is a bit different from the support forms. Those forms submit directly to front where this one submits to the Django backend. We could just create an email using the Front email backend (currently, this uses the normal email backend) but then every single time somebody fills out this form, there will be an email from the default Front user to `support@`. Maybe that's ok?